action.yml

```diff
@@ -0,0 +1,73 @@
+name: Auto Semver
+description: Auto generate semver
+inputs:
+  major_labels:
+    description: space-separated labels indicate it would bump major version
+    required: false
+    default: major
+  minor_labels:
+    description: space-separated labels indicate it would bump minor version
+    required: false
+    default: minor
+  patch_labels:
+    description: space-separated labels indicate it would bump patch version
+    required: false
+    default: patch
+  token:
+    description: the repo's Github token that use the action
+    required: false
+    default: ${{ github.token }}
+  repository:
+    description: the repo that user the action
+    required: false
+    default: ${{ github.repository }}
+  pr_number:
+    description: the PR number that the pr from
+    required: true
+  previous_semver:
+    description: the previous semver that increment from
+    required: true
+runs:
+  using: composite
+  steps:
+    - id: get_labels
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ inputs.token }}
+        GH_REPO: ${{ inputs.repository }}
+        PR_NUMBER: ${{ inputs.pr_number }}
+      run: |
+        # this would return multiple labels, since pull request may attach multiple labels and one of the label may indicate the predefined semver label
+        labels=$(gh pr view $PR_NUMBER --json labels \
+        | jq -r '.labels[].name' | xargs echo)
+        echo "::set-output name=labels::$labels"
+    - id: semver
+      uses: swaglive/action-semver@v1
+      with:
+        input: ${{ inputs.previous_semver }}
+    - id: calculate_semver
+      shell: python
+      run: |
+
+        labels = '${{ steps.get_labels.outputs.labels }}'
+        input_major_labels = '${{ inputs.major_labels }}'
+        input_minor_labels = '${{ inputs.minor_labels }}'
+        input_patch_labels = '${{ inputs.patch_labels }}'
+
+        labels = set(labels.split())
+
+        major = int(${{ steps.semver.outputs.major }})
+        minor = int(${{ steps.semver.outputs.minor }})
+        patch = int(${{ steps.semver.outputs.patch }})
+
+        major = major + bool (set(input_major_labels.split()) & labels);
```

Comment:
The space between `bool (` should be removed

Reply:
I think you can write it as normal if statement.
```python
if set(input_major_labels.split()).intersection(labels):
    major += 1

if set(input_minor_labels.split()).intersection(labels):
    minor += 1

if set(input_patch_labels.split()).intersection(labels):
    patch += 1
```

a the trailing `;` should be removed.

Reply:
ok, I have remove trailing `;` and finally I prefer less line solution `bool()` one